infrastructure/ansible/roles/traffic-router/defaults/main.yml

```diff
@@ -126,36 +126,44 @@ tr_cache_health_json_refresh_period: 1000
 tr_cache_config_json: "{{tr_db_dir}}/cr-config.json"
 tr_cache_config_json_refresh_period: 60000
 
-# log4j.properties
-tr_log4j_opts:
-  - "# ACCESS is set to be a Syslog Appender"
-  - log4j.appender.ACCESS=org.apache.log4j.RollingFileAppender
-  - "log4j.appender.ACCESS.File={{tr_log_dir}}/access.log"
-  - log4j.appender.ACCESS.MaxFileSize=200MB
-  - log4j.appender.ACCESS.MaxBackupIndex=10
-  -
-  - "# ACCESS uses PatternLayout."
-  - log4j.appender.ACCESS.layout=org.apache.log4j.PatternLayout
-  - log4j.appender.ACCESS.layout.ConversionPattern=%m%n
-  - log4j.appender.ACCESS.threshold=INFO
-  -
-  - log4j.appender.A1=org.apache.log4j.RollingFileAppender
-  - "log4j.appender.A1.file={{tr_log_dir}}/traffic_router.log"
-  - log4j.appender.A1.maxFileSize=100MB
-  -
-  - "# A1 uses PatternLayout."
-  - log4j.appender.A1.layout=org.apache.log4j.PatternLayout
-  - log4j.appender.A1.layout.ConversionPattern=%-5p %d{yyyy-MM-dd'T'HH:mm:ss.SSS} [%t] %c - %m%n
-  - log4j.appender.A1.threshold=ALL
-  -
-  - "# Set root logger level to DEBUG and its only appender to A1."
-  - "# Valid levels are: ALL, TRACE, DEBUG, INFO, WARN, ERROR, FATAL, NONE"
-  - log4j.rootLogger=WARN, A1
-  -
-  - "# Set application logger levels"
-  - log4j.logger.org.apache.traffic_control.traffic_router=INFO
-  - log4j.logger.org.apache.traffic_control.traffic_router.core.access=INFO, ACCESS
-  - log4j.additivity.org.apache.traffic_control.traffic_router.core.access=false
+# log4j2.xml
+tr_log4j2_opts: |
+  <Configuration>
+      <Appenders>
+          <Console name="stdout" target="SYSTEM_OUT">
+              <PatternLayout pattern="%m%n" />
+          </Console>
+          <RollingFile name="traffic_router_access"
+                       fileName="{{tr_log_dir}}/access.log"
+                       filePattern="{{tr_log_dir}}/access-%d{yyyy-MM-dd}.log" >
+              <PatternLayout pattern="%m%n" />
+              <Policies>
+                  <SizeBasedTriggeringPolicy size="200MB" />
+              </Policies>
+              <DefaultRolloverStrategy max="10" />
+              <ThresholdFilter level="INFO" />
+          </RollingFile>
+          <RollingFile name="traffic_router"
+                       fileName="{{tr_log_dir}}/traffic_router.log"
+                       filePattern="{{tr_log_dir}}/traffic_router-%d{yyyy-MM-dd}.log" >
+              <Policies>
```

Comment:
`main.yml` needs a PatternLayout:
https://github.com/apache/trafficcontrol/blob/9a1ab1c99601136810e85533ebc286d9779e3976/traffic_router/core/src/main/conf/log4j2.xml#L33-L35

Reply:
:( what is happening.. it is there in main/log4j2.xml and i copied from main/log4j2.xml, how it is missing..